Remove copy_imag and copy_real methods

anjali411 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39099 Add view_as_real for complex tensors
* **#39065 Remove copy_imag and copy_real methods**

Differential Revision: [D21803939](https://our.internmc.facebook.com/intern/diff/D21803939)

anjali411 committed:
Remove copy_imag and copy_real methods

[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

albanD commented on aten/src/ATen/native/ReduceOps.cpp:
```diff
@@ -681,15 +681,15 @@ static Tensor &std_var_out(Tensor &result, const Tensor &self, IntArrayRef dim,
   if (at::isComplexType(self.scalar_type())){
     ScalarType dtype = c10::toValueType(get_dtype(result, self, {}, true));
-    Tensor real_in = self.copy_real().to(dtype);
+    Tensor real_in = at::real(self);
```
You lost the `.to(dtype)` here, is that expected? Why was it needed before?
Same for the others below.

anjali411 replied:
`copy_real` and `copy_imag` methods returned a complex tensor (`re+0j` / `im+0j`) before where the returned tensor's imaginary part was always zero and so `.to(equivalent_float_type)` was just added to return a float tensor ( as we are doing with `real` and `imag` now)

albanD replied:
Sounds good!

albanD reviewed (commented):
Code deletion 🎉

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"




[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"


Differential Revision: [D21803939](https://our.internmc.facebook.com/intern/diff/D21803939)

[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"


Differential Revision: [D21803939](https://our.internmc.facebook.com/intern/diff/D21803939)

[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"


Differential Revision: [D21803939](https://our.internmc.facebook.com/intern/diff/D21803939)

[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"


Differential Revision: [D21803939](https://our.internmc.facebook.com/intern/diff/D21803939)

[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"


Differential Revision: [D21803939](https://our.internmc.facebook.com/intern/diff/D21803939)

[ghstack-poisoned]

anjali411 committed:
Update on "Remove copy_imag and copy_real methods"


Differential Revision: [D21803939](https://our.internmc.facebook.com/intern/diff/D21803939)

[ghstack-poisoned]
